src/Colopl/Spanner/Connection.php

```diff
@@ -443,4 +462,83 @@ protected function runQueryCallback($query, $bindings, Closure $callback)
 
         return $result;
     }
+
+    /**
+     * Returns current mode
+     *
+     * @return string
+     */
+    protected function getSessionNotFoundMode()
+    {
+        return $this->config['sessionNotFoundErrorMode'] ?? self::CLEAR_SESSION_POOL;
+    }
+
+    /**
+     * Handle "session not found" errors
+     *
+     * @template T
+     * @param  Closure(): T $callback
+     * @return T
+     * @throws InvalidArgumentException|NotFoundException|AbortedException
+     */
+    protected function sessionNotFoundWrapper(Closure $callback)
+    {
+        $handlerMode = $this->getSessionNotFoundMode();
+        if (empty($handlerMode) || $this->sessionPool === null) {
+            // skip handlers
+            return $callback();
+        }
+
+        if (!in_array($handlerMode, [
+                self::MAINTAIN_SESSION_POOL,
+                self::CLEAR_SESSION_POOL,
+            ])
+        ) {
+            throw new InvalidArgumentException("Unsupported sessionNotFoundErrorMode [{$handlerMode}].");
+        }
+        try {
+            return $callback();
+        } catch (NotFoundException $e) {
+            // ensure if this really error with session
+            if ($this->causedBySessionNotFound($e)) {
+                if ($this->inTransaction()) {
+                    // if we inside transaction then throw abort exception
+                    throw new AbortedException(self::SESSION_NOT_FOUND_CONDITION);
```

Comment:
This removes all context information given by `$e`. I think `$e` should be passed to AbortedException.

Reply:
fixed in force-pushed pr